app/routes.py

```diff
@@ -1,2 +1,277 @@
+    task.completed_at = None
+
+    db.session.commit()
+
+    return jsonify({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            }
+        }), 200
+
+from app.models.goal import Goal
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+
+@goals_bp.route("", methods = ["POST", "GET"])
+def handle_goals():
+    if request.method == "POST":
+    
+        request_body = request.get_json()
+
+        if "title" not in request_body:
+            return jsonify({
+                "details": "Invalid data"
+            }), 400
+        else:
+            new_goal = Goal(title = request_body["title"])
+            db.session.add(new_goal)
+            db.session.commit()
+            
+        return jsonify({
+                "goal": {
+                    "id": new_goal.goal_id,
+                    "title": new_goal.title
+                }
+            }), 201
+
+    elif request.method == "GET":
+        goal_query = request.args.get("title")
+
+        if goal_query:
+            goals = Goal.query.filter_by(title = goal_query)
+        else:
+            goals = Goal.query.all()
+        
+        list_of_goals = []
+        for goal in goals:
+            list_of_goals.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+
+        if list_of_goals is None:
+            return jsonify([]), 200
+        else:
+            return jsonify(list_of_goals), 200
+
+@goals_bp.route("/<goal_id>", methods = ["GET", "PUT", "DELETE"])
+def handle_specific_goals(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    
+
+    if goal is None:
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        return {
+            "goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}, 200
+
+    elif request.method == "PUT":
+        updated_body = request.get_json()
+
+        goal.title = updated_body["title"]
+        db.session.commit()
+
+        return jsonify({
+            "goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }), 200
+        
+
+@goals_bp.route("/<goal_id>/tasks", methods = ["POST", "GET"])
+def handle_tasks_in_goals(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return jsonify(None), 404
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        task_ids = request_body["task_ids"]
+
+        for task_id in task_ids:
+            task = Task.query.get(task_id)
+            task.goal_id = goal.goal_id
```

Comment:
This is absolutely a valid way to associate a task with a goal. But we can also think about using either end of the goal → tasks relationship we created in the models to help.

One approach that's essentially equivalent to what is written here would be
```python
            task.goal = goal  # use actual model instances rather than the raw ids. `goal` is added to the task by the db.relationship
```

Or to go from the opposite side of the relationship, we could write
```python
            goal.tasks.append(task)  # append the looked up task to the tasks collection of the goal. tasks is added to the goal by the db.relationship
```

In either situation, we should also think about whether to interpret POSTing to this endpoint as setting the exact list of tasks that should be a part of the goal, or whether we are adding thes tasks to the goal (in addition to anything that was already there). Neither the project description nor the test `test_post_task_ids_to_goal_already_with_goals` are entirely clear on the desired behavior. Which perspective does your implementation take? How could we implement the other approach?